Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/MfaHdiAdditionalClusterDetail.java

```diff
@@ -41,17 +43,17 @@ public MfaHdiAdditionalClusterDetail(String clusterName, String userName, String
     @Nullable
     @Override
     public synchronized String getTenantId(){
-        if(this.tenantId != null){
+        if(StringUtils.isNoneBlank(this.tenantId)){
             return this.tenantId;
         }
 
         try {
-            String location = SparkBatchSubmission.getInstance().negotiateAuthMethod(getConnectionUrl());
-            if (location != null && location.matches(SparkBatchSubmission.probeLocationPattern)) {
-                Matcher m = Pattern.compile(SparkBatchSubmission.probeLocationPattern).matcher(location);
-                if (m.find()) {
-                    this.tenantId = m.group("tenantId");
-                }
+            String location = SparkBatchSubmission.getInstance().probeAuthRedirectUrl(getConnectionUrl());
+            if (location != null) {
+                this.tenantId = Arrays.stream(URI.create(location).getPath().split("/"))
```

Comment:
Can you add a sample `location` string in the comment here? It would help us understand how we parse tenant Id from `location` string

Reply:
np